micrometer-observation/src/main/java/io/micrometer/observation/aop/ObservedAspect.java

```diff
@@ -0,0 +1,211 @@
+ * }
+ * </pre>
+ *
+ * @author Jonatan Ivanov
+ * @since 1.10.0
+ */
+@Aspect
+@NonNullApi
+public class ObservedAspect {
+
+    private static final String DEFAULT_OBSERVATION_NAME = "method.observed";
+
+    private static final Predicate<ProceedingJoinPoint> DONT_SKIP_ANYTHING = pjp -> false;
+
+    private final ObservationRegistry registry;
+
+    @Nullable
+    private final Observation.KeyValuesProvider<ObservedAspectContext> keyValuesProvider;
+
+    private final Predicate<ProceedingJoinPoint> shouldSkip;
+
+    public ObservedAspect(ObservationRegistry registry) {
+        this(registry, null, DONT_SKIP_ANYTHING);
+    }
+
+    public ObservedAspect(ObservationRegistry registry,
+            Observation.KeyValuesProvider<ObservedAspectContext> keyValuesProvider) {
+        this(registry, keyValuesProvider, DONT_SKIP_ANYTHING);
+    }
+
+    public ObservedAspect(ObservationRegistry registry, Predicate<ProceedingJoinPoint> shouldSkip) {
+        this(registry, null, shouldSkip);
+    }
+
+    public ObservedAspect(ObservationRegistry registry,
+            @Nullable Observation.KeyValuesProvider<ObservedAspectContext> keyValuesProvider,
+            Predicate<ProceedingJoinPoint> shouldSkip) {
+        this.registry = registry;
+        this.keyValuesProvider = keyValuesProvider;
+        this.shouldSkip = shouldSkip;
+    }
+
+    @Around("@within(io.micrometer.observation.annotation.Observed)")
+    @Nullable
+    public Object observeClass(ProceedingJoinPoint pjp) throws Throwable {
+        if (shouldSkip.test(pjp)) {
+            return pjp.proceed();
+        }
+
+        Method method = ((MethodSignature) pjp.getSignature()).getMethod();
+        Observed observed = getDeclaringClass(pjp).getAnnotation(Observed.class);
+        return observe(pjp, method, observed);
+    }
+
+    @Around("execution (@io.micrometer.observation.annotation.Observed * *.*(..))")
+    @Nullable
+    public Object observeMethod(ProceedingJoinPoint pjp) throws Throwable {
+        if (shouldSkip.test(pjp)) {
+            return pjp.proceed();
+        }
+
+        Method method = getMethod(pjp);
+        Observed observed = method.getAnnotation(Observed.class);
+        return observe(pjp, method, observed);
+    }
+
+    private Object observe(ProceedingJoinPoint pjp, Method method, Observed observed) throws Throwable {
+        String name = observed.name().isEmpty() ? DEFAULT_OBSERVATION_NAME : observed.name();
+        Signature signature = pjp.getStaticPart().getSignature();
+        String contextualName = observed.contextualName().isEmpty() ? getContextualName(signature)
+                : observed.contextualName();
+
+        Observation observation = Observation.createNotStarted(name, new ObservedAspectContext(pjp), registry)
+                .contextualName(contextualName).lowCardinalityKeyValue("class", signature.getDeclaringTypeName())
```

Comment:
maybe this `class` and `method` should be coming from a `DocumentedObservation`? Also maybe it should be `class.name` and `class.method` or sth like that - cause `method` can be an HTTP method? Or am I exaggerating?

Reply:
I'm taking a look at using the DocumentedObservation here, `class.name` maybe `method.name` makes sense we need to check the naming conventions, since right now it is `class` and `method` in `TimedAspect`.

Reply:
fixed